### PR TITLE
Chapter 13

### PR DESCRIPTION
**Pull request by nuriman1**

### CODE REVIEW

**Comment by nuriman1** [comment]
ready

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
   include SessionsHelper
+
+  private
+
+  def logged_in_user
+   return unless logged_in?
+
+   store_location
+   flash[:danger] = "Please log in."
+   redirect_to login_url
```
Comment (saobv-2223) [inline app/controllers/application_controller.rb]:
```suggestion
    return unless logged_in?

    store_location
    flash[:danger] = "Please log in."
    redirect_to login_url
```

File: app/controllers/microposts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  @micropost = current_user.microposts.build micropost_params
+  @micropost.image.attach params[:micropost][:image]
+  if @micropost.save
+    flash[:success] = "Micropost created!"
+    redirect_to root_url
+ else
+    @feed_items = current_user.feed.paginate page: params[:page]
+    render "static_pages/home"
+  end
+end
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]:
```suggestion
  def create
    @micropost = current_user.microposts.build micropost_params
    @micropost.image.attach params[:micropost][:image]
    if @micropost.save
      flash[:success] = "Micropost created!"
      redirect_to root_url
    else
      @feed_items = current_user.feed.paginate page: params[:page]
      render "static_pages/home"
    end
  end
```

File: app/controllers/microposts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  def destroy
+    if @micropost.destroy
+      flash[:success] = "Micropost deleted"
+      redirect_to request.referrer || root_url
+    else
+      flash[:danger] = "Micropost not deleted"
+      redirect_to root_url
+  end
+end
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb:22]:
```suggestion
    end
  end
```

File: app/controllers/microposts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+private
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]:
```suggestion
  private
```

File: app/controllers/static_pages_controller.rb [hunk 5]
```diff
@@ -1,5 +1,10 @@
 class StaticPagesController < ApplicationController
-  def home; end
+  def home 
+    return if logged_in?
+
+    @micropost = current_user.microposts.build
+    @feed_items = current_user.feed.paginate(page: params[:page]).per `number`
```
Comment (saobv-2223) [inline app/controllers/static_pages_controller.rb]:
```suggestion
    @feed_items = current_user.feed.paginate(page: params[:page]).per 5
```
Don't forget to move 5 to setting.yml file and call it here

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -10,6 +10,7 @@ def index
   
   def show
     @user = User.find_by id: params[:id]
+    @microposts = @user.microposts.paginate(page: params[:page])
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    @microposts = @user.microposts.paginate(page: params[:page]).per 5
```

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (nuriman1) [reply app/controllers/application_controller.rb]: done

File: app/controllers/microposts_controller.rb (hunk 2 above)
Comment (nuriman1) [reply app/controllers/microposts_controller.rb]: done

File: app/controllers/microposts_controller.rb (hunk 3 above)
Comment (nuriman1) [reply app/controllers/microposts_controller.rb:22]: done

File: app/controllers/static_pages_controller.rb (hunk 5 above)
Comment (nuriman1) [reply app/controllers/static_pages_controller.rb]: done

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (nuriman1) [reply app/controllers/users_controller.rb]: done

**Comment by nuriman1** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**